### ISSUE TITLE
SSL errors are not generating permanent error

### ISSUE THREAD

**Issue by bigon** [corecode/dma#65]
Hi,

In case of error, `smtp_init_crypto()` returns 1, this means that `deliver_to_host()` will return 1 to `deliver_remote()` which is translated as a temporary failure.

Shouldn't failure to establish a secure SSL connection be a permanent failure?

**Comment by corecode** [comment]
No, if the server as a temporary certificate issue, we shouldn't bounce the mail.

**Comment by corecode** [comment]
What do you think?
